immuneML/encodings/pdbencoding/PDBEncoder.py

```diff
@@ -0,0 +1,250 @@
+            counter = 0
+
+            for model in parserObject:
+                for chain in model:
+                    residueCounter = 0
+                    for residue in chain:
+                        if residueCounter == 0:
+                            list.append(residue.get_parent())
+                            residueCounter = 1
+
+                        for atom in residue:
+                            if (atom.get_name() == "CA"):
+
+                                if "FULL_SEQUENCE" in region:
+                                    list.append(atom.get_coord())
+
+                                else:
+                                    if "IMGT_CDR3" in region and (isInCDR3(atom.full_id[3][1]) or len(collection) >= 2):
+                                        list.append(atom.get_coord())
+
+                                    if "IMGT_CDR2" in region and (isInCDR2(atom.full_id[3][1]) or len(collection) >= 2):
+                                        list.append(atom.get_coord())
+
+                                    if "IMGT_CDR1" in region and (isInCDR1(atom.full_id[3][1]) or len(collection) >= 2):
+                                        list.append(atom.get_coord())
+
+
+                    saveList = list.copy()
+                    collection.append(saveList)
+                    counter = counter + 1
+                    list.clear()
+
+            lightID = -1
+            heavyID = -1
+            antigenID = -1
+
+            for i in range(0,len(collection)):
+                chainID = collection[i][0].id
+
+                if 'L' in chainID or 'A' in chainID:
+                    lightID = i
+
+                elif 'H' in chainID or 'B' in chainID:
+                    heavyID = i
+
+                elif 'P' in chainID or 'C' in chainID:
+                    antigenID = i
+
+                collection[i].pop(0)
+
+
+            if lightID >= 0 and heavyID >= 0 and antigenID == -1:
+                if lightID + heavyID == 1:
+                    antigenID = 2
+
+                elif lightID + heavyID == 3:
+                    antigenID = 0
+
+            elif lightID == -1  or heavyID == -1 or antigenID == -1:
+                lightID = 0
+                heavyID = 1
+                antigenID = 2
+
+
+            lightChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[lightID])))
+
+            caluculateDistance(collection, lightID, antigenID, lightChainDistanceToAntigen)
+
+            heavyChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[heavyID])))
+            caluculateDistance(collection, heavyID, antigenID, heavyChainDistanceToAntigen)
+
+            paddedArrays = paddingOfArray(lightChainDistanceToAntigen,heavyChainDistanceToAntigen)
+
+            structures.append([paddedArrays[0],paddedArrays[1]])
+            CDRcounter = CDRcounter +1
+
+
+        longestAntigen = 0
+        longestChain = 0
+
+        for x in structures:
```

Comment:
here fore example, I can imagine it could be useful to have a PDBStructure object where you can call .get_antigen_length() or .get_antigen_start_stop_indices() or something like that.